packages/EasyStandard/src/Rector/ExplicitBoolCompareRector.php

```diff
@@ -67,50 +68,46 @@ public function getNodeTypes(): array
     }
 
     /**
-     * {@inheritdoc}
+     * @param If_|ElseIf_ $node
      */
     public function refactor(Node $node): ?Node
     {
-        /** @var \PhpParser\Node\Stmt\If_|\PhpParser\Node\Stmt\ElseIf_ $ifNode */
-        $ifNode = $node;
-
-        $conditionNode = $ifNode->cond;
+        $conditionNode = $node->cond;
         $isNegated = false;
 
-        if ($ifNode->cond instanceof BooleanNot) {
-            $conditionNode = $ifNode->cond->expr;
+        if ($node->cond instanceof BooleanNot) {
+            $conditionNode = $node->cond->expr;
             $isNegated = true;
         }
 
         if ($this->isStaticType($conditionNode, BooleanType::class) === false) {
             return null;
         }
 
-        $ifNode->cond = $this->getNewConditionNode($isNegated, $conditionNode);
+        $node->cond = $this->getNewConditionNode($isNegated, $conditionNode);
 
-        return $ifNode;
+        return $node;
     }
 
     /**
      * Returns new condition node.
      */
     private function getNewConditionNode(bool $isNegated, Expr $expr): Expr
     {
-        /** @var \PhpParser\Node\Expr\BinaryOp\Identical $identicalExpr */
-        $identicalExpr = $expr;
```

Comment:
This assumption was actually false. It can be anything, e.g. `StaticCall` or `MethodCall` etc.

Better is to add condition for such node, e.g.

```php
if ($expr instanceof Identical) {
   // ...
}
```